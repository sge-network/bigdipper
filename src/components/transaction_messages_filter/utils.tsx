export const getFilterLabels = () => {
  return ([
    {
      key: 'none',
      display: 'none',
    },
    {
      key: 'bank',
      display: 'bank',
    },
    {
      key: 'crisis',
      display: 'crisis',
    },
    {
      key: 'distribution',
      display: 'distribution',
    },
    {
      key: 'governance',
      display: 'governance',
    },
    {
      key: 'slashing',
      display: 'slashing',
    },
    {
      key: 'staking',
      display: 'staking',
    },
    {
      key: 'authz',
      display: 'authz',
    },
    {
      key: 'feegrant',
      display: 'feegrant',
    },
    {
      key: 'vesting',
      display: 'vesting',
    },
    {
      key: 'profiles',
      display: 'profiles',
    },
    {
      key: 'bet',
      display: 'bet',
    },
    {
      key: 'dvm',
      display: 'dvm',
    },
    {
      key: 'incentivependulum',
      display: 'incentive pendulum',
    },
    {
      key: 'sportevent',
      display: 'sport event',
    },
    {
      key: 'strategicreserve',
      display: 'strategic reserve',
    },
    {
      key: 'others',
      display: 'others',
    },
  ]);
};
