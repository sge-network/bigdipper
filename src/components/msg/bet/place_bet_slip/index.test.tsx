import React from 'react';
import { RecoilRoot } from 'recoil';
import renderer from 'react-test-renderer';
import { MockTheme } from '@tests/utils';
import { MsgPlaceBetSlip } from '@models';
import PlaceBetSlip from '.';

// ==================================
// mocks
// ==================================

jest.mock('@components', () => ({
  Name: (props) => <div id="Name" {...props} />,
}));

// ==================================
// unit tests
// ==================================
describe('screen: TransactionDetails/MsgPlaceBetSlip', () => {
  it('matches snapshot', () => {
    const message = new MsgPlaceBetSlip({
      category: 'bet',
      type: 'MsgPlaceBetSlip',
      fromAddress: 'fromAddress',
      bets: [{
        uid: 'abcd-12345678-efgh',
        amount: [{
          denom: 'usge',
          amount: '200000000',
        }],
        ticket: 'test_ticket1',
      }],
    });
    const component = renderer.create(
      <RecoilRoot>
        <MockTheme>
          <PlaceBetSlip
            message={message}
          />
        </MockTheme>
      </RecoilRoot>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
