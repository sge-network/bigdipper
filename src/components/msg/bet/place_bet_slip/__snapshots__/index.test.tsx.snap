// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: TransactionDetails/MsgPlaceBetSlip matches snapshot 1`] = `
<div>
  <p
    className="MuiTypography-root MuiTypography-body1"
  >
    <div
      components={
        Array [
          <Name
            address="address"
            name="address"
          />,
          <b />,
        ]
      }
      i18nKey="message_contents:txPlaceBetSlipContentOne"
      id="message_contents:txPlaceBetSlipContentOne"
      values={{}}
    />
  </p>
  <div
    className="makeStyles-placebetslip"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      <div
        components={
          Array [
            <b />,
          ]
        }
        i18nKey="message_contents:txPlaceBetSlipContentTwo"
        id="message_contents:txPlaceBetSlipContentTwo"
        values={
          Object {
            "uid": "1234",
            "amount": "19 SGE",
            "ticket": "test_ticket_1",
          }
        }
      />
    </p>
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      <div
        components={
          Array [
            <b />,
          ]
        }
        i18nKey="message_contents:txMultisendContentTwo"
        id="message_contents:txMultisendContentTwo"
        values={
          Object {
            "uid": "4321",
            "amount": "91 SGE",
            "ticket": "test_ticket_2",
          }
        }
      />
    </p>
  </div>
</div>
`;
